### ISSUE TITLE
Table of content hightlight bug

### ISSUE THREAD

**Issue by nejcr** [Rocketseat/gatsby-themes#33]
**Description of bug**

When navigation through the docs I found that table of content on the right side is not coloring the sections correctly
The other issue is that if you go to 
**To Reproduce**

1. Go to https://rocketdocs.netlify.app/usage/creating-docs
2. Click on 'Line numbers' in the table of contents

**Expected behavior**

I would expect the "Line numbers" to be colored. Instead, the element that is colored(selected) is "react live".
My suspicion is that it always selects the element in the center screen.

**Screenshots**

https://recordit.co/VixKdtCVGX

**Comment by jpedroschmitz** [comment]
Thanks for the report @nejcr!

Our table of contents algorithm is designed to consider an element active if he reaches the middle of the page, that's the reason this "small intended bug" happens. I'll be diving into this problem soon, looking for a better and smart solution.

**Comment by jpedroschmitz** [comment]
Hey, in the last week's I've tried many alternatives to fix this "issue", but, unfortunately, it's not an easy problem.

I've checked many documentation websites, different strategies and all of them have drawbacks. None seems to be perfect. 

That's the reason I will stick with the strategy we are using now. 

In case anyone wants to send a PR or have a solution idea, I'll be glad to reopen this issue and accept a PR. 

Thanks for your patience and understanding 💜
